src/Decompiler/Analysis/SsaState.cs

```diff
@@ -110,12 +110,67 @@ public void Dump(bool trace)
 			}
 		}
 
-		/// <summary>
-		/// Deletes a statement by removing all the ids it references 
-		/// from SSA state, then removes the statement itself from code.
-		/// </summary>
-		/// <param name="pstm"></param>
- 		public void DeleteStatement(Statement stm)
+        [Conditional("DEBUG")]
+        public void CheckUses()
+        {
+            CheckUses(s => Debug.WriteLine(s));
+        }
+
+        public void CheckUses(Action<string> error)
+        {
```

Comment:
Although I like the idea of this method in principle, I'm concerned about its implementation. Is these any way you can perform the check without actually modifying the SSA state? Ideally, checks / validations should be read-only operations, with no side effects on the object being validated.